scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -987,6 +989,62 @@ def publish_visibility(self) -> PublishVisibility:
         return PublishVisibility(visibility) if visibility else None
 
 
+class GenericAssetConfig(Config):
+    """Generic Asset Config class.
+
+    Example:
+        # Remote storage path for asset data
+        remote_path:
+            type: azureblob
+            storage_name: my_storage
+            container_name: my_container
+            container_path: foo/bar
+        # Local folder path to copy remote files into
+        local_path: ./prompt
+    """
+
+    def __init__(self, file_name: Path):
+        """Initialize object for the Generic Asset Properties extracted from storage.yaml.
+
+        Args:
+            file_name (Path): Storage config file to load and validate.
+        """
+        super().__init__(file_name)
+        self._path = None
+        self._validate()
+
+    def _validate(self):
+        """Validate the yaml file."""
+        Config._validate_exists('generic_asset.path', self.remote_path)
+        Config._validate_enum('generic_asset.path.type', self.remote_path.type.value, PathType, True)
+        Config._validate_exists('generic_asset.local_path', self.local_path)

```

Comment:
Will every generic asset have all of these properies?

Reply:
For now, any generic asset that has an extra_config would have exactly these properties.  If that changes in the future, then we will update the validation here.